src/Model/Url.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SitemapPlugin\Model;
+
+use DateTimeInterface;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+
+final class Url implements UrlInterface
+{
+    /** @var string */
+    private $location;
+
+    /** @var DateTimeInterface */
+    private $lastModification;
+
+    /** @var string */
+    private $changeFrequency;
+
+    /** @var float */
+    private $priority;
+
+    /** @var Collection */
+    private $alternatives;
+
+    /** @var Collection|ImageInterface[] */
+    private $images;
+
+    public function __construct(string $location)
+    {
+        $this->setLocation($location);
+        $this->alternatives = new ArrayCollection();
+        $this->images = new ArrayCollection();
+    }
+
+    public function getLocation(): string
+    {
+        return $this->location;
+    }
+
+    public function setLocation(string $location): void
+    {
+        $this->location = $location;
+    }
+
+    public function getLastModification(): ?DateTimeInterface
+    {
+        return $this->lastModification;
+    }
+
+    public function setLastModification(DateTimeInterface $lastModification): void
+    {
+        $this->lastModification = $lastModification;
+    }
+
+    public function getChangeFrequency(): string
+    {
+        return $this->changeFrequency;
+    }
+
+    public function setChangeFrequency(ChangeFrequency $changeFrequency): void
+    {
+        $this->changeFrequency = (string) $changeFrequency;
+    }
+
+    public function getPriority(): ?float
+    {
+        return $this->priority;
+    }
+
+    public function setPriority(float $priority): void
+    {
+        if (0 > $priority || 1 < $priority) {
+            throw new \InvalidArgumentException(\sprintf(
+                'The value %s is not supported by the option priority, it must be a number between 0.0 and 1.0.', $priority
+            ));
+        }
+
+        $this->priority = $priority;
+    }
+
+    /**
+     * @return Collection|UrlAlternativeInterface[]
+     */
+    public function getAlternatives(): Collection
+    {
+        return $this->alternatives;
+    }
+
+    /**
+     * @param UrlAlternativeInterface[] $alternatives
+     */
+    public function setAlternatives(iterable $alternatives): void
```

Comment:
This is basically `addAlternatives` because it doesn't get emptied first.

Reply:
Updated the file. Clearing the collections before adding